src/Raven.Studio/typescript/viewmodels/database/cmpXchg/editCmpXchg.ts

```diff
@@ -318,23 +464,31 @@ class editCmpXchg extends viewModelBase {
             });
     }
 
-    formatValue() {
-        eventsCollector.default.reportEvent("cmpXchg", "format");
-        try {
-            const editorText = this.valueEditor.getSession().getValue();
-            const tempValue = JSON.parse(editorText);
-            const formatted = this.stringify(tempValue);
-            this.valueText(formatted);
-        } catch (e) {
-            messagePublisher.reportError("Could not format json", undefined, undefined, false);
-        }
-    }
-
     updateUrl(valueKey: string) {
         const editUrl = appUrl.forEditCmpXchg(valueKey, this.activeDatabase());
         router.navigate(editUrl, false);
     }
 
+    cloneCmpXch() {
+        eventsCollector.default.reportEvent("cmpXchg", "clone");
+        this.isCreatingNewItem(true);
+        this.key("");
+        this.globalValidationGroup.errors.showAllMessages(false);
+    }
+
+    addMetadata() {
+        eventsCollector.default.reportEvent("cmpXchg", "add-metadata");
+        if (!this.hasMetadata()) {
+            this.hasMetadata(true);
+            this.metadataEditor().contentText(null);
```

Comment:
null or empty string?

Reply:
null, otherwise you see the content text again in ace-editor once you re-open metadata